language/compiler/ir-to-bytecode/syntax/src/syntax.rs

```diff
@@ -1188,26 +1211,55 @@ fn parse_type_formal<'input>(
     }
 }
 
+fn parse_generics<'input, F, R>(
```

Comment:
Hmm. I hadn't thought about this part. Let me think about it a bit. I'm not sure how this might interact with improving the error handling for this kind of common pattern.

Reply:
That's a good point. I personally would use the following pattern for better error messages:
Call the high order function => high order function returns some "local" error => refine the error in the caller, with more knowledge about the context